specification/app/resource-manager/Microsoft.App/preview/2023-11-02-preview/ManagedEnvironmentsSecrets.json

```diff
@@ -0,0 +1,357 @@
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/managedEnvironments/{environmentName}/secrets/{secretName}/getValue": {
+      "post": {
+        "tags": [
+          "ManagedEnvironmentSecrets"
+        ],
+        "summary": "Get secret value for a managedEnvironmentSecret.",
+        "description": "Get secret value for a managedEnvironmentSecret.",
+        "operationId": "ManagedEnvironmentSecrets_GetValue",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "environmentName",
+            "in": "path",
+            "description": "Name of the Environment.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$"
+          },
+          {
+            "name": "secretName",
+            "in": "path",
+            "description": "Name of the Secret.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ManagedEnvironmentSecret"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get secret value for a managedEnvironmentSecret": {
+            "$ref": "./examples/ManagedEnvironmentSecrets_GetValue.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ManagedEnvironmentSecretsCollection": {
+      "description": "Collection of Secrets for Environments",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of secret resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedEnvironmentSecret"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ManagedEnvironmentSecret": {
+      "description": "Secret resource for managedEnvironment.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
```

Comment:
Dear ARM reviewer, after reading ARM doc about 'TrackedReourse vs ProxyResource' and I feel like ProxyResource meets current needs. But I would like your recommendation on this: should I make a resource to be TrackedResource by default? Thanks in advance.

Reply:
Does your resource need a location? Would you want it to be deleted along with its containing resource group? Would you want to assign RBAC permissions on that resource's scope?

Reply:
Thank you @TimLovellSmith. 
One follow-up question please, the doc says both resource types support RBAC permissions. Do you mean there's any undocumented difference regarding RBAC?
![image](https://github.com/Azure/azure-rest-api-specs/assets/82434622/afd46790-ddd1-4c6a-b568-555e5ca6b30f)